File: blog/20110206/spring_data_mongo_repository/src/main/java/fits/sample/DataRepository.java
package fits.sample;

import java.util.List;
import org.springframework.data.document.mongodb.repository.MongoRepository;

public interface DataRepository extends MongoRepository<Data, java.math.BigInteger> {
	//指定名の Data を取得する
	List<Data> findByName(String name);

	//指定の名前を含み、ポイントが指定値より大きい Data を取得する
	List<Data> findByNameLikeAndPointGreaterThan(String name, int point);

}

File: kotlin/spring/webflux/mongodb-sample/src/main/kotlin/sample/repository/DataRepository.kt
package sample.repository

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.mongodb.core.ReactiveMongoTemplate
import org.springframework.stereotype.Repository
import sample.data.Data

@Repository
class DataRepository {
    @Autowired
    lateinit var template: ReactiveMongoTemplate

    fun findData(id: String) = template.findById(id, Data::class.java)
    fun saveData(data: Data) = template.insert(data)
}